library/Fission/Web/Auth.hs

```diff
@@ -3,31 +3,51 @@ module Fission.Web.Auth
   , authWithContext
   , basic
   , mkAuth
+  , handler
   , module Fission.Web.Auth.Class
   , module Fission.Web.Auth.Types
   ) where
 
+import           Fission.Prelude
+import           Fission.Models
+
+import qualified Fission.Platform.Heroku.Auth.Types as Heroku
+
 import           Servant
+import           Servant.Server.Experimental.Auth
+
 import           Database.Esqueleto
 
-import           Fission.Models
-import           Fission.Platform.Heroku.Auth.Types as Heroku
-import           Fission.Prelude
+import           Network.Wai
+
 import           Fission.Web.Auth.Class
 import           Fission.Web.Auth.Types
+import           Fission.Web.Auth.Types as Auth
+import           Fission.Web.Auth.Utils
+import qualified Fission.Web.Auth.Error as Auth
+import qualified Fission.Web.Auth.Basic as Basic
+import qualified Fission.Web.Auth.JWT       as JWT
+
+import qualified Fission.User as User
+import           Fission.User.DID.Types
 
-type Checks = '[BasicAuthCheck (Entity User), BasicAuthCheck Heroku.Auth]
+type Checks = '[AuthHandler Request DID, AuthHandler Request (Entity User), BasicAuthCheck Heroku.Auth]
 
 -- | Construct an authorization context
 mkAuth ::
-  ( MonadAuth (Entity User) m
-  , MonadAuth Heroku.Auth   m
+  ( MonadAuth (AuthHandler Request DID)           m
+  , MonadAuth (AuthHandler Request (Entity User)) m
+  , MonadAuth (BasicAuthCheck Heroku.Auth)        m
   )
   => m (Context Checks)
 mkAuth = do
+  didAuth    <- verify
```

Comment:
Nice

Reply:
Actually, while we’re in here. Something that I’ve found confusing in the test branch is the name `verify`, since it’s not actually running that action; it’s returning a thing that _can_ verify. So why not rename the function to `verifier` or `getVerifier`? Thoughts? Tackle in this PR?

Reply:
yup I like it :+1: